### PR TITLE
[Merged by Bors] - feat: add `Module.finrank_quotient_add_finrank_le`

### PR DESCRIPTION
**Pull request by riccardobrasca**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dimension/Finite.lean [hunk 1]
```diff
@@ -426,6 +427,17 @@ theorem Module.finrank_zero_iff [NoZeroSMulDivisors R M] :
   rw [← rank_zero_iff (R := R), ← finrank_eq_rank]
   norm_cast
 
+/-- Similar to `rank_quotient_add_rank_le` but for `finrank` and a finite `M`. -/
+lemma Module.finrank_quotient_add_finrank_le (N : Submodule R M) :
+    finrank R (M ⧸ N) + finrank R N ≤ finrank R M := by
+  by_cases h : Subsingleton R
+  · rw [finrank_zero_iff.2 (Module.subsingleton R _), finrank_zero_iff.2 (Module.subsingleton R _),
+      finrank_zero_iff.2 (Module.subsingleton R _)]
+  rw [not_subsingleton_iff_nontrivial] at h
```
Comment (acmepjz) [inline Mathlib/LinearAlgebra/Dimension/Finite.lean:436]:
```suggestion
  haveI := nontrivial_of_invariantBasisNumber R
```

File: Mathlib/LinearAlgebra/Dimension/StrongRankCondition.lean [hunk 2]
```diff
@@ -443,6 +443,13 @@ noncomputable instance {R M : Type*} [DivisionRing R] [AddCommGroup M] [Module R
 theorem finrank_eq_rank [Module.Finite R M] : ↑(finrank R M) = Module.rank R M := by
   rw [Module.finrank, cast_toNat_of_lt_aleph0 (rank_lt_aleph0 R M)]
 
+/-- If `M` is finite, then `finrank N = rank N` for all `N : Submodule M`. Note that
+such an `N` can be not finitely generated. -/
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Dimension/StrongRankCondition.lean:447]:
```suggestion
such an `N` need not be finitely generated. -/
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**